### PR TITLE
recording: improvements to screenshot masking

### PR DESCRIPTION
**Pull request by marandaneto**
## :bulb: Motivation and Context
* only mask if there's any content (text, image)
* added missing components that are placeholders for wireframes but not for screenshots

## :pencil: Checklist

- [x] I reviewed the submitted code.
- [ ] I added tests to verify the changes.
- [ ] I updated the docs if needed.
- [x] No breaking change or entry added to the changelog.

### CODE REVIEW

File: PostHog/Replay/PostHogReplayIntegration.swift [hunk 1]
```diff
@@ -171,8 +171,38 @@
...
+            }
+
+            if let theSwitch = view as? UISwitch {
+                if isSwitchSensitive(theSwitch) {
+                    maskableWidgets.append(view.toAbsoluteRect(parent))
+                    return
+                }
+            }
+
+            if let picker = view as? UIPickerView {
```
Comment (marandaneto) [inline PostHog/Replay/PostHogReplayIntegration.swift:204]: Added a few missing items that are placeholders for wireframes but not for screenshots and can be sensitive

File: PostHog/Replay/PostHogReplayIntegration.swift [hunk 2]
```diff
@@ -225,29 +255,63 @@
...
+            }
         }
 
         private func isImageViewSensitive(_ view: UIImageView) -> Bool {
             var isAsset = false
             if let image = view.image {
                 isAsset = isAssetsImage(image)
+            } else {
+                // if there's no image, there's nothing to mask
+                return false
```
Comment (marandaneto) [inline PostHog/Replay/PostHogReplayIntegration.swift:314]: Even if masking is enabled, only mask if there's actually any content (text|image)

**Review by daibhin — APPROVED** [review]
Random thought: I don't recall seeing any exit early conditions. I know rrweb will not check child nodes if a parent has been masked which could help some of the extra processing time here

**Comment by marandaneto** [comment]
> Random thought: I don't recall seeing any exit early conditions. I know rrweb will not check child nodes if a parent has been masked which could help some of the extra processing time here

`findMaskableWidgets` (screenshot masking) returns if any condition is met and doesn't check the children's list, so I mask the whole X-Y no matter what.
`toWireframe` indeed does not do it, but the reason is that a component can be composed of different types, eg a Button contains a Label but I have to cast to both types to read the property of each of them, so it's just easier to let it run thru the whole thing, but it's a place to improve if the wireframe has performance issues as well for sure, good point.

**Pull request merged**